control/server.py

```diff
@@ -204,6 +226,39 @@ def _start_spdk(self):
         for trtype in spdk_transports.split():
             self._create_transport(trtype.lower())
 
+    def _stop_spdk(self):
+        """Stops SPDK process."""
+        assert self.spdk_process is not None # should be verified by the caller
+
+        return_code = self.spdk_process.returncode
+        rpc_socket = self.config.get("spdk", "rpc_socket")
+
+        # Terminate spdk process
+        signal.signal(signal.SIGCHLD, signal.SIG_DFL)
+        if return_code is not None:
+            self.logger.error(f"SPDK({rpc_socket}) pid {self.spdk_process.pid} already terminated, exit code: {return_code}")
+        else:
+            self.logger.info(f"Terminating SPDK({rpc_socket}) pid {self.spdk_process.pid}...")
+            self.spdk_process.terminate()
+        try:
+            timeout = self.config.getfloat("spdk", "timeout")
+            stdout, stderr = self.spdk_process.communicate(timeout=timeout)
+            for pair in [(stdout, "stdout"), (stderr, "stderr")]:
```

Comment:
Also, as I said previously, I believe the stdout and stderr will always be None here, so why bother with this?

Reply:
@trociny 🖖 I think you're right, however I feel uncomfortable treating .communicate() as returning void, ignoring stderr/stdout 😀. this code does not print a thing with current Popen argument, but might produce some useful output if codebase changes. WDYT?

Reply:
I don't see anything wrong with ignoring the communicate return value in this case if we expect it to be always None.